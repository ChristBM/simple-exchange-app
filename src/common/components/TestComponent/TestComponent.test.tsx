/* eslint-disable no-undef */
import fixture from './TestComponent.test.fixture';
import TestComponentPO from './TestComponent.test.po';

describe('<TestComponent />', () => {
  const { subtitle, btnLabel } = fixture;
  let page: TestComponentPO;

  beforeEach(() => {
    page = new TestComponentPO();
  });

  it('Renders correctly (snapshot)', () => {
    // Given + When
    const tree = page.createSnapshot({ subtitle: 'subtitle', btnLabel: 'add' });

    // Then
    expect(tree).toMatchSnapshot();
  });

  it('Should display a subtitle', () => {
    // Given
    page.initialize({ subtitle, btnLabel });

    // When
    const { subtitleText } = page;

    // Then
    expect(subtitleText).toBeInTheDocument();
  });

  it('Should display an add button', () => {
    // Given
    page.initialize({ subtitle, btnLabel });

    // When
    const { addBtn } = page;

    // Then
    expect(addBtn).toBeInTheDocument();
  });

  describe('User interactions', () => {
    it('Should display an add button', async () => {
      // Given
      page.initialize({ subtitle, btnLabel });

      // When
      await page.onAddClickBtn();

      // And
      const { titleText } = page;

      // Then
      expect(titleText).toBeInTheDocument();
    });
  });
});
