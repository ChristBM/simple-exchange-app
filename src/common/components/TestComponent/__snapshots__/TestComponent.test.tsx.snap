// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TestComponent /> Renders correctly (snapshot) 1`] = `
<div
  className="sc-beyTiQ eMKGUP"
>
  <h1
    className="sc-guDMob iwoBUU"
  />
  <p
    className="sc-dmyDmy kSCDqM"
  >
    subtitle
  </p>
  <button
    className="sc-hLQTFJ jTDXOb"
    onClick={[Function]}
  >
    add
  </button>
</div>
`;
